tests/gradients/test_param_shift_hessian.py

```diff
@@ -0,0 +1,453 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the gradients.param_shift_hessian module."""
+
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+class TestParameterShiftHessian:
+    """Tests for the param_shift_hessian method"""
+
+    def test_2term_shift_rules1(self):
+        """Test that the correct hessian is calculated for a single RX operator"""
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, diff_method="parameter-shift", max_diff=2)
+        def circuit(x):
+            qml.RX(x, wires=0)
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        x = np.array(0.1, requires_grad=True)
+
+        jacobian = qml.jacobian(qml.grad(circuit))(x)
```

Comment:
technically this is also the Hessian! Might be good to use variable names along the lines of `hessian` and `expected_hessian`

Reply:
Will do!